File: Project_4_UrbanIndie/Frontend(Angular)/src/app/Shared/Modals/ProductModal.ts
import { ProductReviews } from "./ProductReviews";

export interface ProductModal {

    product_id: String;
    product_usin:String;
	product_title:String;
	product_price:String;
	product_brand:String;
	product_image1:String;
	product_image2:String;
	product_image3:String;
	product_image4:String;
	product_maincategory:String;
	product_category:String;
	product_subcategory:String;
	product_total_rating:String;
	product_average_rating:String;
	product_seller_id:String;
	product_quantity:String;
	product_status:String;
	product_description:String;
    product_color:String[];
    product_size:String[];
    model_details:String[];
    product_reviews:Array<ProductReviews>;
    product_details:String[];
    delievery_options:String[];
}

File: Project_4_UrbanIndie/Backend(Host)/src/main/java/com/backend/ProductMicroservice/modals/ProductModal.java
package com.backend.ProductMicroservice.modals;

import java.util.HashSet;
import java.util.List;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@Document(collection = "ProductCollection")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ProductModal {

    @Id
    private String product_id;
    private String product_usin;
    private String product_mrp;
    private Double product_discount;
    private String product_title;
    private String product_price;
    private String product_brand;
    private String apparelCategory;
    private String productType;
    private String product_total_rating;
    private Double product_average_rating;
    private String product_seller_id;
    private String product_status;
    private String departmentType;
    private String product_description;
    private boolean isAddInWishList;
    private boolean isAddInCart;

    private List<String> product_images;

    @Builder.Default
    HashSet<String> product_color = new HashSet<>();

    @Builder.Default
    HashSet<String> product_size = new HashSet<>();

    @Builder.Default
    HashSet<String> model_details = new HashSet<>();

    @Builder.Default
    HashSet<ProductReviews> product_reviews = new HashSet<>();

    @Builder.Default
    HashSet<String> product_details = new HashSet<>();

    @Builder.Default
    HashSet<String> delievery_options = new HashSet<>();

    @Builder.Default
    HashSet<String> product_tags = new HashSet<>();


}
